### PR TITLE
Update README (example with React Hooks)

### LINKED ISSUE #784: Use React Hooks in README
### Summary:
React Hooks in modern way to create a React components and maybe soon classes components will be deprecated.

### PR DESCRIPTION
**Pull request by nagayev**
Fixes #784 .

Changes proposed:
+ Add hook component
- Delete class component

Acceptance Checklist:
- [x] The commit message follows the guidelines in `CONTRIBUTING.md`.
- [x] Documentation (README.md) and examples have been updated as needed.
- [x] If this is a code change, a spec testing the functionality has been added.
- [x] If the commit message has [changed] or [removed], there is an upgrade path above.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,46 +53,37 @@ const customStyles = {
...
-      modalIsOpen: false
-    };
-
-    this.openModal = this.openModal.bind(this);
-    this.afterOpenModal = this.afterOpenModal.bind(this);
-    this.closeModal = this.closeModal.bind(this);
+function App(){
+  openModal = openModal.bind(this);
+  afterOpenModal = afterOpenModal.bind(this);
+  closeModal = closeModal.bind(this);
```
Comment (diasbruno) [inline README.md]: I think you don't need to `bind` in this case. They can be removed.

File: README.md [hunk 2]
```diff
@@ -53,46 +53,37 @@ const customStyles = {
...
+          isOpen={modalIsOpen}
+          onAfterOpen={afterOpenModal}
+          onRequestClose={closeModal}
           style={customStyles}
           contentLabel="Example Modal"
         >
 
-          <h2 ref={subtitle => this.subtitle = subtitle}>Hello</h2>
-          <button onClick={this.closeModal}>close</button>
+          <h2 ref={_subtitle => subtitle = _subtitle}>Hello</h2>
```
Comment (diasbruno) [inline README.md]: `_subtitle => subtitle = _subtitle` to `_subtitle => (subtitle = _subtitle)`

Comment (diasbruno) [reply README.md]: Also, `subtitle` is missing it variable declaration. It can be on top of the component.

File: README.md (hunk 1 above)
Comment (nagayev) [reply README.md]: @diasbruno hm, you are right.
I forget remove it.

**Comment by nagayev** [comment]
@diasbruno 
Well, i fixed it.
Is it time to squash and push commits?

**Comment by diasbruno** [comment]
Awesome, @nagayev. I'll merge this on Monday. Thanks.

**Comment by diasbruno** [comment]
Just make sure that this branch is rebased with master...so we can squash on merge.

**Pull request merged**